interpreter/cling/include/cling/Interpreter/DynamicLibraryManager.h

```diff
@@ -66,7 +67,9 @@ namespace cling {
     ///
     SearchPathInfos m_SearchPaths;
 
-    InterpreterCallbacks* m_Callbacks;
+    InterpreterCallbacks* m_Callbacks = nullptr;
+
+    Dyld* m_Dyld = nullptr;
```

Comment:
Please use a `unique_ptr<my_Dyld>`.

Reply:
That does not quite work. Doing so requires the ctor and the dtor to be in the same object file which is not the case and we cannot move one or the other. We want the symbol resolution facility to be in its own object file because we inspect the assembly code as it is performance critical.

Reply:
> Doing so requires the ctor and the dtor to be in the same object file

O I see - but I think you meant "the implementation of DynamicLibraryManager's ctor and dtor must see Dyld's declaration" and you don't know how to do that without moving Dyld's class declaration into a header? Then maybe do that - it's probably nicer anyway and prevents having to have a mixture of Dyld and DynamicLibraryManager member functions in the same .o?

> in its own object file because we inspect the assembly code

I don't understand that. Are you saying when coding this you wanted to have an easier time inspecting the code's assembly? OK but that does not preempt moving the class declaration out.

Reply:
> > Doing so requires the ctor and the dtor to be in the same object file
> 
> O I see - but I think you meant "the implementation of DynamicLibraryManager's ctor and dtor must see Dyld's declaration" and you don't know how to do that without moving Dyld's class declaration into a header? 

Yes the definition of Dyld needs to be available in the same object file where the ctor and dtor are.

>Then maybe do that - it's probably nicer anyway and prevents having to have a mixture of Dyld and DynamicLibraryManager member functions in the same .o?
>

Agreed, let's keep it as is.
 
> > in its own object file because we inspect the assembly code
> 
> I don't understand that. Are you saying when coding this you wanted to have an easier time inspecting the code's assembly?

Yes.

> OK but that does not preempt moving the class declaration out.

I would not move the class definition in the header file. On design level, nobody needs to actually know about Dyld and its internal structure -- it only has one user-facing function the rest is performance scaffolding. On performance level, I would not move the class definition in the header file. This would make it harder for the optimizer to do pointer escape analysis. The feedback from the previous review slowed down the code by 30% on average and we have still been unable to track the exact reason. I would be reluctant to repeat the same exercise just to be able to type `std::unique_ptr<Dyld>` for a private data member :)